pkg/collector/autodiscovery/configresolver.go

```diff
@@ -262,6 +250,8 @@ func (cr *ConfigResolver) processDelService(svc listeners.Service) {
 			if err != nil {
 				log.Errorf("Failed to stop check '%s': %s", id, err)
 			}
+			// cleaning up the cache map
```

Comment:
For consistency, could we move this whole for loop in a new `ac.unschedule(checks)` method, even if cr is the only user of it?
That would avoid cr modifying ac's internal state

Reply:
done 👍